crates/turborepo-pidlock/src/lib.rs

```diff
@@ -171,9 +171,14 @@ impl Pidlock {
             Ok(pid) if process_exists(pid) => {
                 Some(pid.try_into().expect("if a pid exists it is a valid u32"))
             }
-            Ok(_) => {
+            Ok(pid) => {
                 warn!("stale pid file at {:?}", self.path);
-                None
+                if let Err(e) = fs::remove_file(&self.path) {
+                    warn!("failed to remove stale pid file: {}", e);
+                    Some(pid.try_into().expect("if a pid exists it is a valid u32"))
```

Comment:
Do we want to return the pid if the owning process is not running? I think this should return None unconditionally

Reply:
If we can't remove the pid file, the daemon is not going to start up, because it requires the file it writes to be a new file. If we return `None` here, we'll try to start the daemon, when at this point, we actually do need to kick an error out to the user (we can't remove a stale pid).

It may be possible that we want a `Result` instead of an `Option` as a return value...

Reply:
+1 on making this return `Result`